### PR TITLE
first pass at a layout/nested widget tutorial

### PR DESCRIPTION
**Pull request by stuartromanek**
looking for feedback of all kinds, including

- what should this type of thing officially be called?
- is there a better way to reference `contextualOnly` and `skipInitialModal` than commenting them out in a code block (i didn't want to create multiple instances of similar configuration for the copy-paste happy)
- grammer / spelling etc lol

### CODE REVIEW

File: tutorials/getting-started/layout-widgets.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+You’ve built a couple of custom widgets, you’ve got a powerful `area` running down the main column of your template, awesome. Now you want start pairing widgets up, creating complex layouts of widgets. `singletons` let you enforce particular widgets in particular pieces of layout, but you want the editor to decide where it's appropriate to create these complex arrangements and be able to mix them with simpler widgets.
+
+You want a widget that creates a layout of inner widgets.
+
+### Create a layout widget
+
+Generally, you need a dedicated widget to create layout. The Apostrophe team sometimes refers to these as ‘layout widgets’ that give form and structure to ‘content widgets’ (widgets whose sole job is to display content). **In this example we'll make a simple two column layout.**
+
+With the `apostrophe-cli` installed
```
Comment (boutell) [inline tutorials/getting-started/layout-widgets.md]: This should be a link. And I'd suggest wording like "We'll use apostrophe-cli to make a new widget quickly."

File: tutorials/getting-started/layout-widgets.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+Now we'll define what can be in a `two-column` widget by defining it in `lib/modules/two-column-widgets/index.js`:
+
+```javascript
+module.exports = {
+  extend: 'apostrophe-widgets',
+  label: 'Two Column Layout',
+  // contextualOnly: true,
+  // skipInitialModal: true
```
Comment (boutell) [inline tutorials/getting-started/layout-widgets.md]: This second one is redundant if we have contextualOnly.

File: tutorials/getting-started/layout-widgets.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  }
+```
+
+Now we'll define what can be in a `two-column` widget by defining it in `lib/modules/two-column-widgets/index.js`:
+
+```javascript
+module.exports = {
+  extend: 'apostrophe-widgets',
+  label: 'Two Column Layout',
+  // contextualOnly: true,
```
Comment (boutell) [inline tutorials/getting-started/layout-widgets.md]: This should be active, not commented out, right?

File: tutorials/getting-started/layout-widgets.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+#### Conveniences for layout widgets (contextualOnly and skipInitialModal)
+
+There are two commented properties on the widget configuration above and they both do a similar thing. These are useful for a widget dedicated to nesting other widgets.
+
+> **contextualOnly: true** If your widget contains *only* other areas and singletons you don't need the typical manager modal UI popping up when you create the widget, nor do you need an Edit button to edit non-existing configuration. `contextualOnly` will shortcut these and instantly plop your empty widget on the page.
+
+<video src="/docs/videos/tutorials/editor/contextualTrue.mp4" autoplay="" loop="true" muted="true"></video>
+
+> **skipInitialModal: true** If you want to skip the widget manager modal when the widget is created (like `contextualOnly`) but preserve the Edit UI for later user, `skipInitialModal` is for you. This is useful for widgets that have secondary configuration, like setting a background color.
```
Comment (boutell) [inline tutorials/getting-started/layout-widgets.md]: Later use, not later user?

File: tutorials/getting-started/layout-widgets.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+### Putting it in the page
+
+Now, like any other widget, you need to have a template render.
```
Comment (boutell) [inline tutorials/getting-started/layout-widgets.md]: What's a template render?

How about:

Now, like any other widget, you need to have a `widget.html` template. In this case we'll use that template to call `apos.area` once for each area and introduce nested widgets.

**Comment by boutell** [comment]
Really minor nitpicks I'd like you to look at first, but this is great.

File: tutorials/getting-started/layout-widgets.md (hunk 2 above)
Comment (stuartromanek) [reply tutorials/getting-started/layout-widgets.md]: This is what I was referencing above .. How do I point out where these properties go without duplicating (and possibly losing someone who copy-pastes the first code block) code? Should I just leave them out of the initial example and then provide truncated examples when each prop is explained?

**Comment by stuartromanek** [comment]
Made all edits requested from @boutell see https://github.com/punkave/apostrophe-documentation/pull/77/commits/366f934974f3d007c598316ee06d0fe78430240e

File: tutorials/getting-started/layout-widgets.md (hunk 2 above)
Comment (boutell) [reply tutorials/getting-started/layout-widgets.md]: I would include the one you almost certainly want, fully activated, in the initial example (contextualOnly), and then give truncated examples or just say "similar to the way contextualOnly is used above, you can also use..."

Comment (stuartromanek) [reply tutorials/getting-started/layout-widgets.md]: yeah i did something like this in the edit https://github.com/punkave/apostrophe-documentation/commit/366f934974f3d007c598316ee06d0fe78430240e#diff-cf5f835f95f84c963b57555d91853bfdR37

**Comment by boutell** [comment]
Boom! Thank you.

**Pull request merged**